Library/MapleBacon/MapleBacon/Resizer.swift

```diff
@@ -6,176 +6,251 @@ import UIKit
+            
+            self.horizontalRatio = size.width / imgSize.width
+            self.verticalRatio = size.height / imgSize.height
+            self.offset = CGSize(width: imgSize.width - size.width, height: imgSize.height - size.height)
+            self.offsetPoint = CGPoint(x: self.offset.width / 2.0, y: self.offset.height / 2.0)
+        }
+        
+        mutating func applyContentMode(contentMode: UIViewContentMode) {
+            switch contentMode {
+            case .scaleToFill, .redraw:
+                self.newSize = CGSize(width: self.currentSize.width * self.horizontalRatio, height: self.currentSize.height * self.verticalRatio)
+                self.offset  = self.calcOffset(fromSize: self.newSize, toSize: self.intendedSize)
+            case .scaleAspectFill:
+                let ratio = max(self.horizontalRatio, self.verticalRatio)
+                newSize = CGSize(width: self.currentSize.width * ratio, height: currentSize.height * ratio)
+                offset = self.calcOffset(fromSize: newSize, toSize: self.intendedSize)
+            case .scaleAspectFit:
+                let ratio = min(self.horizontalRatio, self.verticalRatio)
+                newSize = CGSize(width: self.currentSize.width * ratio, height: self.currentSize.height * ratio)
+                offset = self.calcOffset(fromSize: newSize, toSize: self.intendedSize)
+            case .center:
+                break
+            case .top:
+                self.offsetPoint.y = 0
+            case .bottom:
+                self.offsetPoint.y = self.offset.height
+            case .left:
+                self.offsetPoint.x = 0
+            case .right:
+                self.offsetPoint.x = self.offset.width
+            case .topLeft:
+                self.offsetPoint = CGPoint(x: 0, y: 0)
+            case .topRight:
+                self.offsetPoint = CGPoint(x: self.offset.width, y: 0)
+            case .bottomLeft:
+                self.offsetPoint = CGPoint(x: 0, y: self.offset.height)
+            case .bottomRight:
+                self.offsetPoint = CGPoint(x: self.offset.width, y: self.offset.height)
+            }
+        }
+        
+        private mutating func calcOffset(fromSize size: CGSize, toSize: CGSize) -> CGSize {
+            let size = CGSize(width: size.width - toSize.width, height: size.height - toSize.height)
+            self.offsetPoint = CGPoint(x: size.width / 2.0, y: size.height / 2.0)
+            
+            return size
+        }
+    }
+    
+    /**
+     Inits with the image to transform
+     */
+    public init(image: UIImage) {
+        self.image = image
+    }
+    
+    /**
+     Resize fkt with default content mode and quality
+     */
+    public func resize(toSize size: CGSize, async: Bool = true, completion: ResizerCompletion) {
+        self.resize(toSize: size, contentMode: .scaleToFill, interpolationQuality: .default, async: async, completion: completion)
+    }
+    
+    /**
+     Resize fkt
```

Comment:
If we feel the need to document public API then this should list every parameter by name and what it does